numpy/core/src/umath/loops_hyperbolic.dispatch.c.src

```diff
@@ -0,0 +1,405 @@
+        npyv_f64 c5 = npyv_lut16_f64((const double*)lut16x18 + 6*16, idx);
+        npyv_f64 c6 = npyv_lut16_f64((const double*)lut16x18 + 7*16, idx);
+        npyv_f64 c7 = npyv_lut16_f64((const double*)lut16x18 + 8*16, idx);
+        npyv_f64 c8 = npyv_lut16_f64((const double*)lut16x18 + 9*16, idx);
+        npyv_f64 c9 = npyv_lut16_f64((const double*)lut16x18 + 10*16, idx);
+        npyv_f64 c10 = npyv_lut16_f64((const double*)lut16x18 + 11*16, idx);
+        npyv_f64 c11 = npyv_lut16_f64((const double*)lut16x18 + 12*16, idx);
+        npyv_f64 c12 = npyv_lut16_f64((const double*)lut16x18 + 13*16, idx);
+        npyv_f64 c13 = npyv_lut16_f64((const double*)lut16x18 + 14*16, idx);
+        npyv_f64 c14 = npyv_lut16_f64((const double*)lut16x18 + 15*16, idx);
+        npyv_f64 c15 = npyv_lut16_f64((const double*)lut16x18 + 16*16, idx);
+        npyv_f64 c16 = npyv_lut16_f64((const double*)lut16x18 + 17*16, idx);
+
+        // no need to zerofy nans or avoid FP exceptions by NO_EXC like SVML does
+        // since we're clearing the FP status anyway.
+        npyv_f64 sign = npyv_and_f64(x, npyv_reinterpret_f64_s64(npyv_setall_s64(0x8000000000000000ull)));
+        npyv_f64 y = npyv_sub_f64(npyv_abs_f64(x), b);
+        npyv_f64 r = npyv_muladd_f64(c16, y, c15);
+        r = npyv_muladd_f64(r, y, c14);
+        r = npyv_muladd_f64(r, y, c13);
+        r = npyv_muladd_f64(r, y, c12);
+        r = npyv_muladd_f64(r, y, c11);
+        r = npyv_muladd_f64(r, y, c10);
+        r = npyv_muladd_f64(r, y, c9);
+        r = npyv_muladd_f64(r, y, c8);
+        r = npyv_muladd_f64(r, y, c7);
+        r = npyv_muladd_f64(r, y, c6);
+        r = npyv_muladd_f64(r, y, c5);
+        r = npyv_muladd_f64(r, y, c4);
+        r = npyv_muladd_f64(r, y, c3);
+        r = npyv_muladd_f64(r, y, c2);
+        r = npyv_muladd_f64(r, y, c1);
+        r = npyv_muladd_f64(r, y, c0);
+        r = npyv_or_f64(r, sign);
+        /**
+         * SVML handel |x| > HUGE_THRESHOLD, INF and NaNs by scalar callout as following:
+         * 1. check special cases
+         * 2. generate lookup index for |x| > HUGE_THRESHOLD
+         * 3. check the offset and return value from table otherwise return `+-1`
+         *    depend on the sign of x
+         *
+         * We fallback to C instead of handel rare cases when abs(x) > HUGE_THRESHOLD
+         * via lut (easier). special cases also treat it as rare cases and handel it
+         * through C call to save few cycles for numeric values.
+         */
+        npy_uint64 callout = npyv_tobits_b64(callout_m);
+        if (NPY_UNLIKELY(callout != ((1 << nlanes) - 1))) {
+            double NPY_DECL_ALIGNED(NPY_SIMD_WIDTH) back2c[npyv_nlanes_f64];
+            npyv_store_f64(back2c, x);
+            for (unsigned i = 0; i < npyv_nlanes_f64; ++i) {
+                if ((callout >> i) & 1) {
+                    continue;
+                }
+                back2c[i] = npy_tanh(back2c[i]);
```

Comment:
I think we can get rid of this call to glibc too, just like the single precision case.